core-plugins/src/main/java/co/cask/hydrator/plugin/common/StructuredToAvroTransformer.java

```diff
@@ -66,7 +67,11 @@ public GenericRecord transform(StructuredRecord structuredRecord,
       if (schemaField == null) {
         throw new IllegalArgumentException("Input record does not contain the " + fieldName + " field.");
       }
-      recordBuilder.set(fieldName, convertField(structuredRecord.get(fieldName), schemaField.getSchema()));
+      if (schemaField.getSchema().getType().equals(co.cask.cdap.api.data.schema.Schema.Type.BYTES)) {
```

Comment:
I think this will fail if the field is nullable. You will need to check and get the simple type. An example can be found here: https://github.com/hydrator/date-transform/blob/master/src/main/java/co/cask/hydrator/plugin/DateTransform.java#L106